libs/execution_context/ebpf_maps.c

```diff
@@ -375,51 +416,52 @@ _update_array_map_entry(
     if (key_value >= map->ebpf_map_definition.max_entries)
         return EBPF_INVALID_ARGUMENT;
 
-    uint8_t* entry = &map->data[*key * map->ebpf_map_definition.value_size];
+    uint8_t* value = _get_array_map_value(map, key_value);
     if (data) {
-        memcpy(entry, data, map->ebpf_map_definition.value_size);
+        memcpy(value, data, map->ebpf_map_definition.value_size);
     } else {
-        memset(entry, 0, map->ebpf_map_definition.value_size);
+        memset(value, 0, map->ebpf_map_definition.value_size);
```

Comment:
Maybe add a test to cover this path as it's apparently not being hit.

Reply:
Agree. The thing is some of these locations might be affected by the results of my 'linux functionality' comparison testing results and will definitely add/update new/existing tests to ensure better coverage.  Depending on the scope of the resulting changes, we could split the coverage between this issue and new ones resulting from that comparison.  Thoughts?